changelog/_unreleased/2024-12-12-update-to-doctrine-dbal-4-2.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Update to doctrine/dbal:4.2
+issue: NEXT-39353
+---
+# Core
+* Changed composer dependency version of `doctrine/dbal` from `^3.9` to `^4.2`
+* Removed method `Shopware\Core\Migration\Test\NullConnection::executeUpdate`
+* Removed method `Shopware\Core\Framework\DataAbstractionLayer\DataAbstractionLayerException::databasePlatformInvalid`
+* Changed `Shopware\Core\Framework\DataAbstractionLayer\Dbal\Common\LastIdQuery::__construct` and `Shopware\Core\Framework\DataAbstractionLayer\Dbal\Common\OffsetQuery::__construct` to accept `Shopware\Core\Framework\DataAbstractionLayer\Dbal\QueryBuilder` instead of `Doctrine\DBAL\Query\QueryBuilder`
+* Changed `Shopware\Core\Framework\DataAbstractionLayer\Dbal\Common\IterableQuery::getQuery` to return `Shopware\Core\Framework\DataAbstractionLayer\Dbal\QueryBuilder` instead of `Doctrine\DBAL\Query\QueryBuilder`
+* Changed `Shopware\Core\Framework\DataAbstractionLayer\Doctrine\RetryableQuery::execute` return type to int|string
+* Changed `Shopware\Core\Framework\DataAbstractionLayer\Dbal\ExceptionHandlerInterface::matchException` parameter and return type to `\Throwable` instead of `\Exception`. All relevant implementations have been updated.
+* Changed `Shopware\Core\Framework\DataAbstractionLayer\Search\Parser\ParseResult::addParameter` to accept `Doctrine\DBAL\ParameterType|\Doctrine|Dbal\ArrayParameterType` as `$type` and use `ParameterType::STRING` as default value
+* Changed `Shopware\Core\Framework\DataAbstractionLayer\Doctrine\MultiInsertQueryQueue` to accept `Doctrine\DBAL\ParameterType` as parameter type and to use prepared statements under the hood
+* Added `Shopware\Core\Framework\DataAbstractionLayer\Util\StatementHelper` to simplify binding multiple parameters to a statement
```

Comment:
Why did doctrine drop this? Do we recommend to keep using `StatementHelper` going forward or do we also follow doctrine there?

Reply:
Fine for me 👍